src/SignhostAPIClient/Rest/ErrorHandling/HttpResponseMessageErrorHandlingExtensions.cs

```diff
@@ -67,6 +70,9 @@ public static class HttpResponseMessageErrorHandlingExtensions
 				case HttpStatusCode.BadRequest:
 					throw new BadRequestException(
 						errorMessage);
+				case HttpStatusCode.PaymentRequired:
+					throw new PaymentRequiredException(
```

Comment:
I suggest we make this more specific when out of credits, eg `OutOfCreditsException` instead of bubbling up al status codes. Might also be better to only raise this when the error details is of the type out-of-credits.